vm/src/stdlib/itertools.rs

```diff
@@ -196,13 +195,19 @@ mod decl {
             vm: &VirtualMachine,
         ) -> PyResult {
             let start = match start.into_option() {
-                Some(int) => int.as_bigint().clone(),
-                None => BigInt::zero(),
+                Some(num) => vm.new_pyobj(num),
+                None => vm.new_pyobj(0),
             };
             let step = match step.into_option() {
-                Some(int) => int.as_bigint().clone(),
-                None => BigInt::one(),
+                Some(int) => {
+                    let val: isize = int.try_to_primitive(vm)?;
+                    Some(vm.new_pyref(val.to_usize().unwrap_or(0)))
+                }
+                None => None,
             };
+            if !PyNumber::check(&start, vm) {
```

Comment:
```suggestion
            if !PyNumber::check(&start, vm) || !PyNumber::check(&step, vm) {
```
It seems `step` also need to be checked via `PyNumber::check`.
https://github.com/python/cpython/blob/3.10/Modules/itertoolsmodule.c#L4178-L4182

Reply:
I didn't because it's a PyIntRef. I'll fix it.

Reply:
PyIntRef is always a PyNumber.